PowerEditor/src/NppIO.cpp

```diff
@@ -403,7 +458,7 @@ BufferID Notepad_plus::doOpen(const generic_string& fileName, bool isRecursive,
             vector<generic_string> patterns;
             if (globbing)
             {
-                const TCHAR * substring = wcsrchr(targetFileName.c_str(), TCHAR('\\'));
+                const TCHAR * substring = _tcsrchr(targetFileName.c_str(), TCHAR('\\'));

```

Comment:
Since Notepad++ is Unicode application since long time ago, I think it's better use **w** version functions than **_t** version ones.

Reply:
Ditto above.

There is one reason why even today one should not mix the `TCHAR` definitions with the `WCHAR` funcs/WINAPIs. MS passed a very large circle and returned to the beginning - they have started  (Windows 10 1903+) to support UNICODE also via its ANSI WINAPI by the help of [UTF-8](https://learn.microsoft.com/en-us/windows/apps/design/globalizing/use-utf8-code-page).

It is a pity that they did not implement their UNICODE support via this from the start - then there did not have to be all the wide-variants of functions like `CreateFileW` at all, just the one `CreateFile` as before.

Reply:
> There is one reason why even today one should not mix the `TCHAR` definitions/funcs with the `WCHAR` ones. MS passed a very large circle and returned to the beginning - they have started (Windows 10 1903+) to support UNICODE also via its ANSI WINAPI by the help of [UTF-8](https://learn.microsoft.com/en-us/windows/apps/design/globalizing/use-utf8-code-page).
 
My understanding was the UTF-8 code page automatically turns API calls into their "wide" variant &#x2014; a kind of runtime equivalent to the UNICODE preprocessor def:

> If the ANSI code page is configured for UTF-8, -A APIs typically operate in UTF-8. This model has the benefit of supporting existing code built with -A APIs without any code changes.
>
> https://learn.microsoft.com/windows/apps/design/globalizing/use-utf8-code-page#-a-vs--w-apis

> And we wouldn't need `TCHAR` or `WCHAR`, just `char` only.

That's a technical debt of C++, not Windows.
A `char` is always UTF-16 in a "modern" language like .NET or (since 2009) [even Delphi](https://docwiki.embarcadero.com/RADStudio/Sydney/en/Unicode_in_RAD_Studio).